src/main/java/games/strategy/engine/lobby/server/db/TimedController.java

```diff
@@ -0,0 +1,13 @@
+package games.strategy.engine.lobby.server.db;
+
+import java.time.Instant;
+
+/**
+ * This class allows integration Tests to override the "now time".
+ * This enables us to make Tests execution-time-independent which is a good thing.
+ */
+abstract class TimedController {
```

Comment:
Not suggesting you change this now, but another possibility is to use the _Strategy_ pattern instead of inheritance or additional method overloads (although a _Strategy_ would require an additional constructor):

```java
public interface Clock {
  Instant now();
}

public class SystemClock implements Clock {
  @Override
  public Instant now() {
    return Instant.now();
  }
}

public class BannedMacController {
  private final Clock clock;

  public BannedMacController() {
    this(new SystemClock());
  }

  @VisibleForTesting
  BannedMacController(final Clock clock) {
    this.clock = clock;
  }

  // . . .
}
```

Then each test fixture could pass a mock `Clock` to the controller:

```java
public class BannedMacControllerTest {
  @Mock
  private Clock clock;

  @InjectMocks
  private BannedMacController controller;

  // . . .
}
```

I can easily see `Clock` being required outside this package because we have other tests that depend on wall time.  Maybe when we hit the next such test, that will be the appropriate time to extract such an interface.